### PR TITLE
MGMT-1445: Port image expiration from Python CronJob to Go goroutine

### PR DESCRIPTION
**Pull request by avishayt**

### CODE REVIEW

File: pkg/s3Expirer/s3Expirer.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/filanov/bm-inventory/pkg/s3wrapper"
+	"github.com/jasonlvhit/gocron"
+	"github.com/sirupsen/logrus"
+)
+
+func Run(s3Config *s3wrapper.Config, cronMinutes, deleteMinutes int, log logrus.FieldLogger) error {
```
Comment (filanov) [inline pkg/s3Expirer/s3Expirer.go:13]: we have this package that you can use https://github.com/filanov/bm-inventory/blob/master/pkg/thread/thread.go

Comment (avishayt) [reply pkg/s3Expirer/s3Expirer.go]: Fixed

**Comment by avishayt** [comment]
Ran 38 of 38 Specs in 505.548 seconds
SUCCESS! -- 38 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestSubsystem (505.55s)
PASS
ok  	github.com/filanov/bm-inventory/subsystem	505.578s

File: cmd/main.go [hunk 2]
```diff
@@ -56,7 +57,9 @@ var Options struct {
 	S3Config                    s3wrapper.Config
 	HostStateMonitorInterval    time.Duration `envconfig:"HOST_MONITOR_INTERVAL" default:"8s"`
 	Versions                    versions.Versions
-	UseK8s                      bool `envconfig:"USE_K8S" default:"true"` // TODO remove when jobs running deprecated
+	UseK8s                      bool          `envconfig:"USE_K8S" default:"true"` // TODO remove when jobs running deprecated
+	ImageExpirationInterval     time.Duration `envconfig:"IMAGE_EXPIRATION_INTERVAL" default:"30m"`
+	ImageExpirationMinutes      int           `envconfig:"IMAGE_EXPIRATION_MINUTES" default:"60"`
```
Comment (filanov) [inline cmd/main.go:62]:
```suggestion
	ImageExpirationMinutes      int           `envconfig:"IMAGE_EXPIRATION_MINUTES" default:"60m"`
```

File: cmd/main.go [hunk 3]
```diff
@@ -139,6 +142,20 @@ func main() {
 
 	events := events.NewApi(eventsHandler, logrus.WithField("pkg", "eventsApi"))
 
+	if Options.ImageExpirationMinutes > 0 {
```
Comment (filanov) [inline cmd/main.go:145]: shouldn't it depend on `UseK8s`? 
I don't think it's clear to use `ImageExpirationMinutes` as a parameter that will define if the expirer should run or not.

File: cmd/main.go [hunk 4]
```diff
@@ -16,6 +16,7 @@ import (
 	"github.com/filanov/bm-inventory/internal/bminventory"
 	"github.com/filanov/bm-inventory/internal/cluster"
 	"github.com/filanov/bm-inventory/internal/common"
+	"github.com/filanov/bm-inventory/internal/imageExpirer"
```
Comment (filanov) [inline cmd/main.go:19]: https://blog.golang.org/package-names

**Comment by avishayt** [comment]
Ran 38 of 38 Specs in 445.070 seconds
SUCCESS! -- 38 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestSubsystem (445.08s)
PASS
ok github.com/filanov/bm-inventory/subsystem 445.115s

**Pull request merged**